WaterLog_Backend/Procedures.cs

```diff
@@ -0,0 +1,299 @@
+            
+        }
+
+        public async Task updateSegmentsEventAsync(int id, string status,double inv,double outv)
+        {
+            SegmentEventsController controller = getSegmentsEventsController();
+            SegmentEventsEntry entry = new SegmentEventsEntry();
+            entry.TimeStamp = DateTime.UtcNow;
+            entry.SegmentId = id;
+            entry.FlowIn = inv;
+            entry.FlowOut = outv;
+            entry.EventType = status;
+            await controller.Post(entry);
+
+        }
+
+    
+        public Boolean isLeakage(double first,double second)
+        {
+            double margin = 2;
+            if((first-second) > margin)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        public string[] populateEmail(int sectionid)
+        {
+            SegmentLeaksController controller = getSegmentLeaksController();
+            var leaks = controller.Get().Result.Value;
+            var leak = leaks.Where(sudo => sudo.SegmentId == sectionid).Single();
+
+            string[] template = { "Segment " + leak.SegmentId, getSegmentStatus(leak.SegmentId), leak.Severity, getLeakPeriod(leak), calculateTotalCost(leak).ToString(), calculatePerHourCost(leak).ToString(), calculateLitresPerHour(leak).ToString(), buildUrl(leak.SegmentId) };
+            return template;
+        }
+
+        private string getLeakPeriod(SegmentLeaksEntry leak)
+        {
+            
+            if (((leak.LatestTimeStamp - leak.OriginalTimeStamp).TotalHours) < 1)
+            {
+                return "1";
+            }
+            else
+            {
+                return ((leak.LatestTimeStamp - leak.OriginalTimeStamp).TotalDays.ToString());
+            }
+           
+        }
+
+        public double calculateTotalCost(SegmentLeaksEntry leak)
+        {
+            SegmentEventsController controller = getSegmentsEventsController();
+            var list = controller.Get().Result.Value;
+            var entry = list.Where(inlist => inlist.SegmentId == leak.SegmentId).Last();
+            
+            var timebetween = (leak.LatestTimeStamp - leak.OriginalTimeStamp).TotalHours;
+            if (timebetween < 1)
+            {
+                timebetween = 1;
+            }
+            else
+            {
+                timebetween = (leak.LatestTimeStamp - leak.OriginalTimeStamp).TotalHours;
+            }
+            var perhour = calculatePerHourCost(leak);
+            return (timebetween * perhour);
+        }
+
+        private string buildUrl(int segmentId)
+        {
+            return "https://localhost:3000/segment/" + segmentId;
+        }
+
+        public double calculatePerHourCost(SegmentLeaksEntry leak)
+        {
+            SegmentEventsController controller = getSegmentsEventsController();
+            var list = controller.Get().Result.Value;
+            var entry = list.Where(inlist => inlist.SegmentId == leak.SegmentId).Last();
```

Comment:
Use a db query instead

Reply:
changed structure of how we access data please check, should solve issue